src/AspNet.Security.OAuth.Apple/AppleAuthenticationEvents.cs

```diff
@@ -22,17 +21,15 @@ public class AppleAuthenticationEvents : OAuthEvents
         /// </summary>
         public Func<AppleGenerateClientSecretContext, Task> OnGenerateClientSecret { get; set; } = async context =>
         {
-            var provider = context.HttpContext!.RequestServices!.GetRequiredService<AppleClientSecretGenerator>();
-            context.Options.ClientSecret = await provider.GenerateAsync(context);
+            context.Options.ClientSecret = await context.Options.ClientSecretGenerator.GenerateAsync(context);
```

Comment:
It's probably off-topic for this PR, but we shouldn't mutate options during request handling. Maybe we should add a `ClientSecret` property to `AppleGenerateClientSecretContext` instead of mutating the one in the options.